### ISSUE TITLE
3.2.x Rss feed problems with links to download elements

### ISSUE THREAD

**Issue by may17** [contao/core#7756]
It seems like there is a bug with download items in the generated feed files.

A single item looks like:
```html
<a href="http://www.mydomain.com/contao/main.php?do=maintenance&ref=d30c5f74&amp;file=files/myPdfFile.pdf" title="My PDF File">My PDF File <span class="size">(178,8 KiB)</span>
```
As you can see the pdf file can not downloaded without backend login.

**How to reproduce:**

- Create news archive
- Create new news item
- Create content element download with an pdf file in the new news item
- Create RSS Feed with the options RSS 2 and Export settings Full articles
- Take a look at the generated xml file in the share folder

**Comment by may17** [comment]
I have analysed the problem. The Problem is located here https://github.com/contao/core/blob/support/3.2/system/modules/core/elements/ContentDownload.php#L86 and here https://github.com/contao/core/blob/support/3.2/system/modules/core/elements/ContentDownloads.php#L143

If the rss feed is generated from the backend the \Environment::get('request') returns mydomain.com/contao and so on.

**Comment by leofeyer** [comment]
@contao/developers How do we solve this?

**Comment by Toflar** [comment]
We can't, I guess. At least I have no idea at all.

**Comment by katgirl** [comment]
At News-Feed we define the base URL and we have the UUID from file and can generate the real link. But you can't trakking the request now. 

If we redirect the download to Root-Page and can download a file by referenz from all pages, we can trakking the site and have a right link

**Comment by leofeyer** [comment]
Maybe we can override the request (`Environment::set('request', $feed->baseUrl)`) and restore it after the feed has been generated?

**Comment by leofeyer** [comment]
Fixed in 9c34300f09dad3c41dba4577f57141a559d362a9.
